packages/react-ui-components/src/DateInput/dateInput.tsx

```diff
@@ -176,6 +182,15 @@ export class DateInput extends PureComponent<DateInputProps, DateInputState> {
             },
         );
 
+        const getTimeFormat = () => {
```

Comment:
This should not be a function imho. The result from `getTimeFormat` is used exactly once per render - so it might as well be calculated in place, e.g.:

```typescript
const timeFormat = dateOnly ? false : (is24Hour ? 'HH:m' : 'hh:m A');
```

But, since nested ternaries are awkward, it would also be okay to turn `getTimeFormat` into a private calculated property of the component, e.g.:

```typescript
private get timeFormat(): false | string {
    if (this.props.dateOnly) {
        return false;
    }

    return this.props.is24Hour ? 'HH:m' : 'hh:m A';
}
```

Reply:
I like your suggestion with the private function in the class component. 

I don't rly like nested ternaries and think a function is way more readable than nesting ternaries.

Thanks for your suggestion I will change this.